### PR TITLE
Added support for Telegram multiple messages

### PR DESCRIPTION
**Pull request by jveres**

### CODE REVIEW

**Comment by gojko** [comment]
Hi,

please also update existing telegram tests and add the tests for the features you added before we can merge this https://github.com/claudiajs/claudia-bot-builder/tree/master/spec/telegram

**Comment by jveres** [comment]
Done!

**Comment by stojanovic** [comment]
Thanks, I'll review it during the weekend

File: lib/telegram/reply.js [hunk 1]
```diff
@@ -1,37 +1,63 @@
 'use strict';
 
 const rp = require('minimal-request-promise');
+const REQUEST_THROTTLE = 1000/30;
```
Comment (stojanovic) [inline lib/telegram/reply.js]: @jveres this is because of the rate limit?

File: lib/telegram/reply.js [hunk 2]
```diff
@@ -1,37 +1,63 @@
...
-  };
 
-  return rp.post(`https://api.telegram.org/bot${tlAccessToken}/${method}`, options);
+    sendAll = function () {
+      if (!messages.length) {
+        return Promise.resolve();
+      } else {
+        return sendSingle(messages.shift())
+        .then(() => {
+          return new Promise((resolve) => {
```
Comment (stojanovic) [inline lib/telegram/reply.js]: no need for `()` for a single param

File: lib/telegram/reply.js [hunk 3]
```diff
@@ -1,37 +1,63 @@
...
+            if (!messages.length)
+              resolve();
+            else
+              setTimeout(() => {
+                resolve();
+              }, REQUEST_THROTTLE);
```
Comment (stojanovic) [inline lib/telegram/reply.js]: My suggestion: `setTimeout(resolve, REQUEST_THROTTLE);`
And please use `{}` if you have else block

**Comment by stojanovic** [comment]
@jveres I have a few minor comments but this looks good. Thanks!

File: lib/telegram/reply.js (hunk 1 above)
Comment (jveres) [inline lib/telegram/reply.js]: Yes, max 30/sec.

File: lib/telegram/reply.js (hunk 2 above)
Comment (jveres) [inline lib/telegram/reply.js]: Eslint gave me error without ()

Comment (stojanovic) [inline lib/telegram/reply.js]: Ah, ok, tnx

**Comment by stojanovic** [comment]
Merged, thanks!

**Comment by jveres** [comment]
👍

**Pull request merged**